train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java

```diff
@@ -2,11 +2,27 @@
 
 import hu.bme.mit.train.interfaces.TrainController;
 
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
 public class TrainControllerImpl implements TrainController {
 
 	private int step = 0;
 	private int referenceSpeed = 0;
 	private int speedLimit = 0;
+	private static TrainControllerImpl self;
```

Comment:
The variable name should be more precise

Reply:
- self renamed to **trainControllerObject**.